docs/how-to/Bar-Memory.rst

```diff
@@ -0,0 +1,97 @@
+
+These BARs are a set of 32-bit or 64-bit registers that are used to define the memory space that PCIe devices can access. The CPU and other system devices also use these to access the resources of the PCIe devices. P2P DMA only works when one device can directly access the local BAR memory of another. If the BAR memory is more than the physical addressing limit of the device, the device will not be able to access the remote BAR. 
+
+To handle any BAR access issues that might occur, you need to be aware of the physical address limitations of the devices and understand the :ref:`BAR configuration of AMD GPUs <bar-configuration>`. This information is important when you need to set up additional MMIO apertures for PCIe devices in the system's physical address space.
+
+Handling physical address limitation
+=============================================
+When a system boots, the system BIOS allocates the physical address space for the components in the system, including system memory and MMIO apertures. There are two MMIO apertures: one allocated less than 4 GB of physical address space for 32-bit compatibility, and one more than 4 GB for devices needing more space.
+
+You can control the memory address of the high MMIO aperture from the system BIOS configuration options. This lets you configure the additional MMIO space to align with the physical addressing limit and allows P2P DMA between the devices. For example, if a PCIe device is limited to 44-bit of physical addressing, you should ensure that the MMIO aperture is set below 44-bit in the system physical address space.
+
+There are two ways to handle this:
+
+* Ensure that the high MMIO aperture is within the physical addressing limits of the devices in the system. For example, if the devices have a 44-bit physical addressing limit, set the ``MMIO High Base`` and ``MMIO High size`` options in the BIOS such that the aperture is within the 44-bit address range, and ensure that the ``Above 4G Decoding`` option is Enabled.  
+
+* Enable the Input-Output Memory Management Unit (IOMMU). When the IOMMU is enabled in non-passthrough mode, it will create a virtual I/O address space for each device on the system. It also ensures that all virtual addresses created in that space are within the physical addressing limits of the device. For more information on IOMMU, see :doc:`../conceptual/iommu`. 
+
+.. _bar-configuration:
+
+BAR configuration for AMD GPUs
+================================================
+
+The following table shows how the BARs are configured for AMD GPUs that have a 44-bit physical address and 48-bit GPU virtual address limit. 
```

Comment:
"that have a 44-bit physical address and 48-bit GPU virtual address limit"
I would drop this part.  The BAR configuration is the same regardless of what the GPU's physical addressing limits are.  Also I wouldn't mention the GPU virtual addressing limit here, it will just confuse things as it's not related to this.